src/dotnet/commands/dotnet-install/InstallCommand.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.ShellShimMaker;
+using Microsoft.DotNet.ToolPackageObtainer;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Build.Tasks;
+using NuGet.Configuration;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class InstallCommand
+    {
+        public static int Run(string[] args)
+        {
+            CommandLine.Parser parser = Parser.Instance;
+            ParseResult result = parser.ParseFrom("dotnet install", args);
+
+            result.ShowHelpOrErrorIfAppropriate();
+
+            AppliedOption parseResult = result["dotnet"]["install"]["tool"];
+
+            var packageId = parseResult.Arguments.Single();
+            var packageVersion = parseResult.ValueOrDefault<string>("version");
+
+            FilePath? configFile;
+
+            var configFilePath = parseResult.ValueOrDefault<string>("configfile");
+            configFile = configFilePath != null
+                ? new FilePath(configFilePath)
+                : GetConfigFileProbingCurrentDirectory();
+
+            var framework = parseResult.ValueOrDefault<string>("framework");
+
+            var executablePackagePath = new DirectoryPath(new CliFolderPathCalculator().ExecutablePackagesPath);
+
+            var toolConfigurationAndExecutableDirectory = ObtainPackage(
+                packageId,
+                packageVersion,
+                configFile,
+                framework,
+                executablePackagePath);
+
+            DirectoryPath executable = toolConfigurationAndExecutableDirectory
+                .ExecutableDirectory
+                .WithSubDirectories(
+                    toolConfigurationAndExecutableDirectory
+                        .Configuration
+                        .ToolAssemblyEntryPoint);
+
+            var shellShimMaker = new ShellShimMaker.ShellShimMaker(executablePackagePath.Value);
+            var commandName = toolConfigurationAndExecutableDirectory.Configuration.CommandName;
+            shellShimMaker.EnsureCommandNameUniqueness(commandName);
+
+            shellShimMaker.CreateShim(
+                executable.Value,
+                commandName);
+
+            EnvironmentPathFactory
+                .CreateEnvironmentPathInstruction()
+                .PrintAddPathInstructionIfPathDoesNotExist();
+
+            Reporter.Output.WriteLine(
+                $"{Environment.NewLine}The installation succeeded. If there is no other instruction. You can type the following command in shell directly to invoke: {commandName}");
```

Comment:
@KathleenDollard customer facing string -- this is shown after the installation. If tools path does not exist on env:path. It will still show, but after the instruction. It will be something like:

tools path does not exist on, please run echo.....

The installation succeeded. If there is no other instruction. You can type the following command in shell directly to invoke demo